examples/gno.land/r/demo/tests/z2_filetest.gno

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"fmt"
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+	"gno.land/r/demo/tests"
+)
+
+func main() {
+	var (
+		eoa        = testutils.TestAddress("someone")
+		rTestsAddr = std.DerivePkgAddr("gno.land/r/demo/tests")
+	)
+	std.TestSetOrigCaller(eoa)
+	// When a single realm in the frames, PrevRealm returns the user
+	if addr := tests.GetPrevRealm().Addr(); addr != eoa {
+		fmt.Printf("want tests.GetPrevRealm().Addr==%s, got %s\n", eoa, addr)
+	}
+	// When 2 or more realms in the frames, PrevRealm returns the second to last
+	if addr := tests.GetRSubtestsPrevRealm().Addr(); addr != rTestsAddr {
+		fmt.Printf("want GetRSubtestsPrevRealm().Addr==%s, got %s\n", rTestsAddr, addr)
+	}
```

Comment:
Since these are both filetests, they're already matched against an "expected" output, so I'd say you can simplify this to the following:

```
println("tests.GetPrevrealm().Addr()", tests.GetPrevRealm().Addr())
println("tests.GetRSubtestsPrevRealm().Addr()", tests.GetRSubtestsPrevRealm().Addr())
```

And then put the expected result in `// Output:`. Same for the following test.

Reply:
Done for z2. For z3_filetest.gno, I think that the intention in the test is a bit lost, which may lead to incomprehension later:
```
func main() {
	var (
		eoa        = testutils.TestAddress("someone")
		rTestsAddr = std.DerivePkgAddr("gno.land/r/demo/tests")
	)
	std.TestSetOrigCaller(eoa)
	println("tests.GetPrevRealm().Addr: ", tests.GetPrevRealm().Addr())
	println("EOA: ", eoa)
	println("tests.GetRSubtestsPrevRealm().Addr(): ", tests.GetRSubtestsPrevRealm().Addr())
	println(`std.DerivePkgAddr("gno.land/r/demo/tests"): `, std.DerivePkgAddr("gno.land/r/demo/tests"))
}

// Output:
// tests.GetPrevRealm().Addr:  g1xufrdvnfk6zc9r0nqa23ld3tt2r5gkyvw76q63
// EOA:  g1wdhk6et0dej47h6lta047h6lta047h6lrnerlk
// tests.GetRSubtestsPrevRealm().Addr():  g1gz4ycmx0s6ln2wdrsh4e00l9fsel2wskqa3snq
// std.DerivePkgAddr("gno.land/r/demo/tests"):  g1gz4ycmx0s6ln2wdrsh4e00l9fsel2wskqa3snq
```

Whereas the following is clear enough:

```
func main() {
    var (
        eoa        = testutils.TestAddress("someone")
        rTestsAddr = std.DerivePkgAddr("gno.land/r/demo/tests")
    )
    std.TestSetOrigCaller(eoa) 
    // Contrarily to z2_filetest.gno we EXPECT GetPrevRealms != eoa (#1704)
    if addr := tests.GetPrevRealm().Addr(); addr != eoa {
        fmt.Printf("want tests.GetPrevRealm().Addr==%s, got %s\n", eoa, addr)
    }                     
    // When 2 or more realms in the frames, it is also different
    if addr := tests.GetRSubtestsPrevRealm().Addr(); addr != rTestsAddr {
        fmt.Printf("want GetRSubtestsPrevRealm().Addr==%s, got %s\n", rTestsAddr, addr)
    }
}

// Output:
// want tests.GetPrevRealm().Addr==g1wdhk6et0dej47h6lta047h6lta047h6lrnerlk, got g1xufrdvnfk6zc9r0nqa23ld3tt2r5gkyvw76q63
```

```

So my current PR will keep using current version (using `println` instead of `fmt.Printf`).
Tell me if it's really needed to change this and I'll do it for the z3_ testfile